Rename PartitionedOutputBuffer and PartitionedOutputBufferManager

kewang1024 opened the pull request:
Rename PartitionedOutputBuffer and PartitionedOutputBufferManager to
OutputBuffer and OutputBufferManager.

Because PartitionedOutputBuffer is not "partitioned" but a general output buffer 
which has three types: partitioned, broadcast and arbitrary. Similarly for 
PartitionedOutputBufferManager.

xiaoxmeng reviewed (approved):
@kewang1024 thanks for the refactor. Please have @mbasmanova to take a look before land.

mbasmanova commented on velox/exec/tests/PartitionedOutputBufferManagerTest.cpp:
```diff
@@ -38,7 +38,7 @@ class PartitionedOutputBufferManagerTest : public testing::Test {
   void SetUp() override {
     pool_ = facebook::velox::memory::addDefaultLeafMemoryPool();
-    bufferManager_ = PartitionedOutputBufferManager::getInstance().lock();
+    bufferManager_ = OutputBufferManager::getInstance().lock();
```
Should we rename the test as well? 

PartitionedOutputBufferManagerTest.cpp -> OutputBufferManagerTest.cpp

mbasmanova commented on velox/core/QueryConfig.h:
```diff
@@ -355,7 +355,7 @@ class QueryConfig {
   // this. The Drivers are resumed when the buffered size goes below
-  // PartitionedOutputBufferManager::kContinuePct % of this.
+  // OutputBufferManager::kContinuePct % of this.
   uint64_t maxPartitionedOutputBufferSize() const {
```
Do we need to rename this property as well?

maxPartitionedOutputBufferSize -> maxOutputBufferSize

mbasmanova reviewed (commented):
@kewang1024 Shorter name is nicer. Is this change backwards-compatible? Will it break Prestissimo?

Please, update PR title and description for clarity. Use active verbs.

Rename PartitionedOutputBufferManager to OutputBufferManager.

kewang1024 commented:
> @kewang1024 Shorter name is nicer. Is this change backwards-compatible? Will it break Prestissimo?

I already have the prestissimo change at hand, but forgot that this would break internally
So now I added back the `PartitionedOutputBufferManager` with `VELOX_ENABLE_BACKWARD_COMPATIBILITY`

I changed the `maxOutputBufferSize` but didn't change the config name `kMaxPartitionedOutputBufferSize` because prestissimo depends on it as well, plan to change that in following PR to keep the risk small
